asyncio_run_in_process/_child.py

```diff
@@ -70,11 +72,69 @@ def update_state_finished(to_parent: BinaryIO, finished_payload: bytes) -> None:
     to_parent.flush()
 
 
+async def _do_task_cleanup(*tasks: 'asyncio.Future[Any]') -> None:
```

Comment:
It might be a good idea to have a section on asyncio to https://github.com/ethereum/snake-charmers-tactical-manual/. I'm _pretty sure_ that this code is correct but it's very non-obvious, it's probably not the first thing a future version of us will try to write in the future. Some kind of documentation on known-good ways to work with Tasks would be useful!